clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntity.java

```diff
@@ -250,4 +301,129 @@ private EhcacheEntityResponse invokeInternal(EhcacheEntityMessage message, boole
     }
   }
 
+  private String getMessageOp(EhcacheEntityMessage message) {
+    switch (message.getType()) {
+      case SERVER_STORE_OP:
+        try {
+          return message.getType() + "/" + getServerStoreOp(message.getOpCode());
+        } catch (IllegalArgumentException e) {
+          return message.getType() + "/" + message.getOpCode();
+        }
+      case LIFECYCLE_OP:
+        try {
+          return message.getType() + "/" + ((LifecycleMessage)message).operation();
+        } catch (ArrayIndexOutOfBoundsException e) {
+          return message.getType() + "/" + message.getOpCode();
+         }
+      default:
+        return message.getType() + "/" + message.getOpCode();
+    }
+  }
+
+  /**
+   * Describes the timeouts for {@link EhcacheClientEntity} operations.  Use
+   * {@link #builder()} to construct an instance.
+   */
+  public static final class Timeouts {
+    private final TimeoutDuration readOperationTimeout;
+    private final TimeoutDuration mutativeOperationTimeout;
+    private final TimeoutDuration lifecycleOperationTimeout;
+
+    private Timeouts(TimeoutDuration readOperationTimeout, TimeoutDuration mutativeOperationTimeout, TimeoutDuration lifecycleOperationTimeout) {
+      this.readOperationTimeout = readOperationTimeout;
+      this.mutativeOperationTimeout = mutativeOperationTimeout;
+      this.lifecycleOperationTimeout = lifecycleOperationTimeout;
+    }
+
+    public TimeoutDuration getReadOperationTimeout() {
+      return readOperationTimeout;
+    }
+
+    public TimeoutDuration getMutativeOperationTimeout() {
+      return mutativeOperationTimeout;
+    }
+
+    public TimeoutDuration getLifecycleOperationTimeout() {
+      return lifecycleOperationTimeout;
+    }
+
+    public static Builder builder() {
+      return new Builder();
+    }
+
+    @Override
+    public String toString() {
+      return "Timeouts{" +
+          "readOperationTimeout=" + readOperationTimeout +
+          ", mutativeOperationTimeout=" + mutativeOperationTimeout +
+          ", lifecycleOperationTimeout=" + lifecycleOperationTimeout +
+          '}';
+    }
+
+    /**
+     * Constructs instances of {@link Timeouts}.  When obtained from
+     * {@link Timeouts#builder()}, the default values are pre-set.
+     */
+    public static final class Builder {
+      private TimeoutDuration readOperationTimeout = TimeoutDuration.of(5, TimeUnit.SECONDS);
```

Comment:
I thought we had decided that the default timeout would be `NONE`?  In any case, if we are to have a default I think it would be better if it's sourced from the config code (as much as is practical).